### PR TITLE
Setup 23.9.0 branch

### PR DESCRIPTION
**Pull request by zoxpx**
**What this PR does / why we need it**:

Setting up boilerplate files for `operator 23.9.0` release

**Which issue(s) this PR fixes** (optional)
Closes # -

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by zoxpx** [comment]
Hey folks (@piyush-nimbalkar, @nikita-bhatia) -- can I please get someone to approve this PR, so that we can working on the next `operator` release?
* there's nothing special on this branch, just the exact same boilerplate changes like we always do..

File: deploy/olm-catalog/portworx/23.9.0/portworx-certified.clusterserviceversion.yaml [hunk 1]
```diff
@@ -0,0 +1,544 @@
+apiVersion: operators.coreos.com/v1alpha1
+kind: ClusterServiceVersion
+metadata:
+  name: portworx-operator.v23.9.0
+  namespace: placeholder
+  annotations:
+    capabilities: Auto Pilot
+    categories: "Storage"
+    description: Cloud native storage solution for production workloads
+    containerImage: registry.connect.redhat.com/portworx/openstorage-operator@sha256:0766693109a9ccb8da5809c68c68c6ba1be77bada3466e789279e1b4e383efb3
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.9.0/portworx-certified.clusterserviceversion.yaml:10]: Let's change this to `docker.io/portworx/px-operator:23.9.0-dev`

File: deploy/olm-catalog/portworx/23.9.0/portworx-certified.clusterserviceversion.yaml [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+  replaces: portworx-operator.v23.7.0
+  maintainers:
+  - name: Portworx
+    email: support@portworx.com
+  provider:
+    name: Portworx
+  keywords: ["portworx", "persistent storage", "storage", "cloud native", "open source"]
+  relatedImages:
+    - name: portworx-operator
+      image: registry.connect.redhat.com/portworx/openstorage-operator@sha256:0766693109a9ccb8da5809c68c68c6ba1be77bada3466e789279e1b4e383efb3
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.9.0/portworx-certified.clusterserviceversion.yaml:65]: Same here

File: deploy/olm-catalog/portworx/23.9.0/portworx-certified.clusterserviceversion.yaml [hunk 3]
```diff
@@ -0,0 +1,544 @@
...
+                  - labelSelector:
+                      matchExpressions:
+                      - key: name
+                        operator: In
+                        values:
+                        - portworx-operator
+                    topologyKey: kubernetes.io/hostname
+              containers:
+              - name: portworx-operator
+                image: registry.connect.redhat.com/portworx/openstorage-operator@sha256:0766693109a9ccb8da5809c68c68c6ba1be77bada3466e789279e1b4e383efb3
```
Comment (piyush-nimbalkar) [inline deploy/olm-catalog/portworx/23.9.0/portworx-certified.clusterserviceversion.yaml:222]: Same here

**Review by piyush-nimbalkar — APPROVED** [review]
LGTM after the CSV fixes.

**Comment by zoxpx** [comment]
Travis checks passed, merging this PR

**Pull request merged**